src/ngx_http_lua_kong_log.c

```diff
@@ -84,8 +85,41 @@ ngx_http_lua_kong_get_dynamic_log_level(ngx_uint_t current_log_level)
     return g_dynamic_log_level;
 }
 
+
 int
-ngx_http_lua_kong_ffi_get_dynamic_log_level(int current_log_level)
+ngx_http_lua_kong_ffi_get_dynamic_log_level(ngx_http_request_t *r,
+    int* current_log_level, int* timeout, int* original_log_level)
 {
-    return ngx_http_lua_kong_get_dynamic_log_level(current_log_level);
+    if (current_log_level == NULL || timeout == NULL || original_log_level == NULL) {
+        return NGX_ERROR;
+    }
+
+    *original_log_level = r == NULL ?
+                          ngx_cycle->log->log_level:
+                          r->connection->log->log_level;
+
+    /* nginx's log_level may be 0x7ffffff0, 0x80000000
+     * see src/core/ngx_log.h
+     */
+    if (*original_log_level > NGX_LOG_DEBUG) {
+        *original_log_level = NGX_LOG_DEBUG;
+    }
+
+    /* timeout, disable the dynamic log level */
+    if (g_dynamic_log_level_timeout_at < ngx_time()) {
```

Comment:
```suggestion
    if (g_dynamic_log_level_timeout_at <= ngx_time()) {
```

I think we should change this check (and for `ngx_http_lua_kong_get_dynamic_log_level`). Otherwise, if `g_dynamic_log_level_timeout_at == ngx_time()`, we will return `original_log_level != `current_log_level` with a `timeout = 0`, breaking the API documented behavior.

Reply:
cc @ADD-SP to comment since he probably wrote the original check.

Reply:
I think we can wrap this logic into a macro to avoid further mistakes.

Reply:
done.